jwql/website/apps/jwql/data_containers.py

```diff
@@ -767,15 +789,18 @@ def get_preview_images_by_instrument(inst):
     # Parse the results to get the rootnames
     filenames = [result['filename'].split('.')[0] for result in results]
 
-    # Get list of all preview_images
-    preview_images = glob.glob(os.path.join(PREVIEW_IMAGE_FILESYSTEM, '*', '*.jpg'))
+    # Get list of all preview_images. Text file contains only preview
+    # images for a single instrument.
+    preview_list_file = f"{PREVIEW_IMAGE_LISTFILE}_{inst.lower()}.txt"
+    preview_images = retrieve_filelist(os.path.join(PREVIEW_IMAGE_FILESYSTEM, preview_list_file))
 
+    # DO WE NEED TO DO THIS??? I GUESS THIS GETS RID OF ANY PREVIEW IMAGES
+    # THAT DON'T MATCH UP TO OBSERVATION FILES.
     # Get subset of preview images that match the filenames
-    preview_images = [os.path.basename(item) for item in preview_images if
-                      os.path.basename(item).split('_integ')[0] in filenames]
+    #preview_images = [os.path.basename(item) for item in preview_images if
```

Comment:
I'm not sure about this. I guess we do still need to filter the instrument-specific list of preview image filenames by file roots from the list of observations?

Reply:
This is interesting. I would say we could tie this up with the new issue we plan to open for the other commented code. Do you think this could be a hold over from the pre-launch filesystem? I know we had a lot of file mismatches prior to launch with old pre-flight data hanging around that wasn't in MAST, right?

Reply:
Thinking about this more, I suspect we should keep it. This is going to remove any preview images that don't match up to a base name in the list of fits files in the filesystem. I think that makes sense to do in that, if we have a preview image file for a file that is no longer in the filesystem, or some non-standard preview image, it will be ignored. I think that's desirable. It's not really clear to me off-hand how a preview image like that would get into the filesystem to begin with barring manual creation.

Reply:
I can see this line taking a long time to run down the road when we have boatloads of data. I would put this towards the top of a list of code to revisit if we need to speed up the web app in the future.